Python/peephole.c

```diff
@@ -536,3 +527,45 @@ PyCode_Optimize(PyObject *code, PyObject* consts, PyObject *names,
     PyMem_Free(codestr);
     return code;
 }
+
+/* Perform basic peephole optimizations to components of a code object.
+   The consts object should still be in list form to allow new constants
+   to be appended.
+
+   To keep the optimizer simple, it bails when the lineno table has complex
+   encoding for gaps >= 255.
+
+   Optimizations are restricted to simple transformations occurring within a
+   single basic block.  All transformations keep the code size the same or
+   smaller.  For those that reduce size, the gaps are initially filled with
+   NOPs.  Later those NOPs are removed and the jump addresses retargeted in
+   a single pass. */
+
+PyObject *
+PyCode_Optimize(PyObject *code, PyObject* consts, PyObject *names,
+                PyObject *lnotab_obj)
+{
+    PyObject* bytecode = NULL;
```

Comment:
The declaration of `bytecode` can be moved into the loop.

Reply:
I am returning bytecode outside the loop, so that would need to be placed inside as well and I prefer not to do it in case the function needs some cleanup in the future.